lib/matplotlib/tests/test_collections.py

```diff
@@ -1149,3 +1149,42 @@ def test_check_masked_offsets():
 
     fig, ax = plt.subplots()
     ax.scatter(unmasked_x, masked_y)
+
+
+@check_figures_equal(extensions=["png"])
+def test_masked_set_offsets(fig_ref, fig_test):
+    x = np.ma.array([1, 2, 3, 4, 5], mask=[0, 0, 1, 1, 0])
+    y = np.arange(1, 6)
+
+    ax_test = fig_test.add_subplot()
+    scat = ax_test.scatter(x, y)
+    x += 1
```

Comment:
```suggestion
```

This confused me down below when seeing `[2, 3, 6], [1, 2, 5]` and wondering why they were offset from one another.